### ISSUE TITLE
Add `Action` based `Match` (or similarly named) method

### ISSUE THREAD

**Issue by TheJayMann** [domn1995/dunet#64]
I am attempting to replace my custom T4 based union generator with dunet.  Currently, my implementation generates an `Either` method which is functionally identical to the `Match` method generated by dunet, and a `Do` method, which is similar, but takes an `Action` for each union case, and does not return a value.  Dunet does not appear to currently have such a method.  The generated code should be identical to the current `Match`, except that it would return `void` rather than `TMatchOutput`, would take `Action` instead of `Func` delegates, and would not need to be generic on `TMatchOutput`, as there would be no output.

**Comment by domn1995** [comment]
Hi, thanks for creating this issue. I've run into this scenario myself, and I think it makes total sense to support it.

One alternative I can suggest is the meantime is create a `Unit` struct and return that instead of `void` for your match functions. It would look something like this:

```cs
public readonly record struct Unit;

[Union]
public partial record SomeUnion
{
   public partial record Foo
   public partial record Bar; 
}

var union = new SomeUnion.Foo();

union.Match(
   foo => 
   { 
      Console.WriteLine("foo"); 
      return new Unit(); 
   },
   bar => 
   { 
      Console.WriteLine("bar"); 
      return new Unit(); 
   }
);
```

Definitely a little more verbose, but might unblock until the feature gets implemented :)

**Comment by TheJayMann** [comment]
I've done similar in the past.  For now, the way I decided to work around it is that I instead returned a different `Action` delegate for each match case, then invoked the action returned.

Anonymized example:
```csharp
            union.Match<Action<Input>>(
                case1 => input => input.Value1 = case1.Value,
                case2 => input => input.Value2 = case2.Value,
                case3 => input => input.Value3 = case3.Value
            ).Invoke(input);
```

**Comment by domn1995** [comment]
Just pushed v1.4.0 which should address your use case, @TheJayMann! Let me know how it works for you :)

Thanks again for filing this issue!

**Comment by TheJayMann** [comment]
Looking at the generated code, it appears that it would work well enough.  I noticed that a corresponding task version is not created.  Although this is not a feature I am seeking currently, I believe it would make sense to implement for the sake of symmetry.

**Comment by domn1995** [comment]
Totally agree. I'll spin that off into its own issue 👍
